src/clientlib/cl_subscription_manager.c

```diff
@@ -1447,13 +1447,24 @@ cl_sm_get_server_socket_filename(cl_sm_ctx_t *sm_ctx, const char *module_name, c
     snprintf(pid_str, 20, "%d", getpid());
     strncat(path, pid_str, PATH_MAX - strlen(path) - 1);
 
+#ifdef HAVE_MKSTEMPS
     /* append temporary file name part */
     strncat(path, ".XXXXXX.sock", PATH_MAX - strlen(path) - 1);
     fd = mkstemps(path, 5);
     if (-1 != fd) {
         close(fd);
         unlink(path);
     }
+#else
+    /* append temporary file name part */
+    strncat(path, ".XXXXXX", PATH_MAX - strlen(path) - 1);
+    fd = mkstemp(path);
+    if (-1 != fd) {
+        close(fd);
+        unlink(path);
+    }
+    strncat(path, ".sock", PATH_MAX - strlen(path) - 1);
```

Comment:
I think that when you add ".sock" to the path here, it does not actually change the opened file, why should it? That means that ".XXXXXX" file will be created and opened, but socket_path will be ".XXXXXX.sock" so it will fail to find the socker or am I mistaken? That is why I suggested to use mktemp, not mkstemp.